docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,403 @@
+
+[id="{p}-beat-custom-configuration"]
+=== Customize Beat configuration
+
+The Beat configuration is provided through the `config` element:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: heartbeat-quickstart
+spec:
+  type: heartbeat
+  version: {version}
+  elasticsearchRef:
+    name: elasticsearch-quickstart
+  config:
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["elasticsearch-quickstart-es-http.default.svc:9200"]
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+----
+
+Alternatively, it can be provided indirectly via a Secret specified in the `configRef` element:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: heartbeat-quickstart
+spec:
+  type: heartbeat
+  version: 7.8.0
+  elasticsearchRef:
+    name: elasticsearch-quickstart
+  configRef:
+    secretName: heartbeat-config
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: heartbeat-config
+stringData:
+  beat.yml: |-
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["elasticsearch-quickstart-es-http.default.svc:9200"]
+----
+
+For more details about Beats configuration, see the link:https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[Beats documentation].
+
+[id="{p}-beat-deploy-elastic-beat"]
+=== Deploy Elastic Beat
+
+ECK supports the deployment of the following Elastic Beats:
+
+- Filebeat
+- Metricbeat
+- Heartbeat
+- Journalbeat
+- Auditbeat
+- Packetbeat
+
+You can specify the Beat to deploy and its version through `type` and `version` elements. ECK computes the correct path to the container image, so the `image` element doesn't have to be provided. ECK creates a new user in Elasticsearch with a minimal set of appropriate roles and permissions to enable the use all Beat features.
+
+[id="{p}-beat-deploy-community-beat"]
+=== Deploy community Beat
+
+ECK supports the deployment of any community Beat. As with Elastic Beats, `type` and `version` specification elements have to be provided. In addition:
+
+- `image` must point to the image to be deployed
+-  role must exist and have the permissions required by the Beat. The role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_kafkabeat_role`.
```

Comment:
Reading this it is not clear to me what your refer to with `role`. Maybe some more explanation?

Reply:
```suggestion
-  a role must exist in Elasticsearch and have the permissions required by the Beat. The role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_kafkabeat_role`.
```
Maybe?